packages/react/utils/src/createContext.tsx

```diff
@@ -18,13 +18,13 @@ export function createContextObj<ContextValueType extends object>(rootComponentN
 
   function Provider(props: ContextValueType & { children: React.ReactNode }) {
     const { children, ...providerProps } = props;
-    // Only re-memoize when prop values change
-    const value = React.useMemo(
-      () => providerProps,
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-      Object.values(providerProps)
-    ) as ContextValueType;
-    return <Context.Provider value={value}>{children}</Context.Provider>;
+    const providerPropsString = JSON.stringify(providerProps);
+    // Only re-memoize when stringified props change
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    const context = React.useMemo(() => ({} as ContextValueType), [providerPropsString]);
+    // mutate properties to ensure closures are up to date
+    Object.entries(providerProps).forEach(([prop, value]) => ((context as any)[prop] = value));
```

Comment:
I think I'm gonna need some help understanding how this actually works.

Reply:
Ah I think I need to do that `JSON.stringify` filter thing we did with Jonathan the other day. Perhaps there is another way to achieve this without `JSON.stringify` though. Let's have a call when you're here so I can explain what's needed here